cms/forms/wizards.py

```diff
@@ -158,15 +158,23 @@ def create_page_titles(page, title, languages):
                 create_title(language, title, page)
 
     @staticmethod
-    def get_first_placeholder(page):
+    def get_placeholder(page, slot=None):
         """
-        Returns the first editable, non-static placeholder or None.
+        Returns the named placeholder or the first editable, non-static
+        placeholder or None.
         """
-        for placeholder in page.get_placeholders():
-            if not placeholder.is_static and placeholder.is_editable:
-                return placeholder
-        else:
-            return None
+        placeholders = page.get_placeholders()
+
+        if slot:
```

Comment:
Also I think that if placeholder is defined then maybe we shouldn't fallback to the first placeholder if the defined one does not exist.
This is because if defined, it means programmer is being explicit about where to add plugins.